client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -26,6 +31,70 @@ const CancelPurchaseButton = React.createClass( {
 		};
 	},
 
+	handleCancelPurchaseClick() {
+		if ( isDomainRegistration( this.props.purchase ) ) {
+			return this.goToCancelConfirmation();
+		}
+
+		this.setState( {
+			showDialog: true
+		} );
+	},
+
+	closeDialog( action ) {
+		if ( action === 'delete' ) {
```

Comment:
Instead of including both dialog actions in the same handler, we can pass `this.submitCancelPurchase` to the `onClick` property of the primary button in the `buttons` array below. This will also prevent the need to check an `action` string entirely.